### ISSUE TITLE
Under certain circumstances NVDA fails to start reading automatically a newly loaded page in Microsoft Edge

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#12258]
### Steps to reproduce:
1. Update Microsoft Edge to its latest stable release (v89).
2. Open Microsoft Edge and load the website of NV Access (it does not occur only with it, but it is a good example).
3. Activate the "Download" link from the menu on the main page.

### Actual behavior:
On the newly loaded "Download" page, NVDA says only "loading complete" and does not start reading the page automatically, even though the corresponding option in NVDA's settings in the "Browse mode" category is checked.

### Expected behavior:
NVDA should automatically start reading the loaded page, if the corresponding option in NVDA's settings in the "Browse mode" category is checked.

### System configuration
- Operating system: Windows 10 Pro version 20H2 (build 19042.867), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-22222,2e37433e, installed, in bulgarian.
- Microsoft Edge version: 89.0.774.63, official build, 64-bit, in bulgarian.
- Processor: Intel Core i5-9400F at 2.90GHz.
- RAM Memory: 16.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC892 at Intel Cannon Point PCH.

#### Does the issue still occur after restarting your computer?
Yes, it does.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - 2020.4 and the result is the same.

#### If add-ons are disabled, is your problem still occurring?
Yes, it is.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes, I did - it did not fix the issue.

**Comment by Brian1Gaff** [comment]
Is not this an old problem on many browsers and can happen due to delays 
somewhere. .
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Kostadin Kolev" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Thursday, April 01, 2021 11:42 AM
Subject: [nvaccess/nvda] Under certain circumstances NVDA fails to start 
reading automatically a newly loaded page in Microsoft Edge (#12258)

**Comment by k-kolev1985** [comment]
No, it is not an old problem. It started with Edge v89.

**Comment by k-kolev1985** [comment]
@Adriani90 Yes, I can still reproduce this with Edge 112 and last alpha of NVDA.

**Comment by Adriani90** [comment]
This is related to virtual cdocument not being refreshed correctly in chromium browsers.
Here are the errors I get when pressing nvda+f5 to see whether NVDA updated its virtual document:

```
IO - inputCore.InputManager.executeGesture (12:42:44.137) - winInputHook (13092):
Input: kb(laptop):NVDA+f5
DEBUG - virtualBuffers.VirtualBuffer._loadBuffer (12:42:44.140) - NVDAObjects.IAccessible.chromium.VirtualBuffer.loadBuffer (13612):
Buffer load took 0.000 sec, 0 chars
DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:44.147) - Dummy-14 (2896):
Thread 21128, build\x86_64\vbufBackends\gecko_ia2\gecko_ia2.cpp, GeckoVBufBackend_t::render, 1442:
Could not get IAccessible2, returning

DEBUGWARNING - virtualBuffers.VirtualBuffer._loadBufferDone (12:42:44.153) - MainThread (18632):
Empty buffer. Waiting for documentLoadComplete event instead
IO - inputCore.InputManager.executeGesture (12:42:47.837) - winInputHook (13092):
Input: kb(laptop):upArrow
IO - speech.speech.speak (12:42:47.849) - MainThread (18632):
Speaking ['Leer']
DEBUGWARNING - virtualBuffers.VirtualBufferTextInfo._getNVDAObjectFromOffset (12:42:47.857) - MainThread (18632):
Couldn't get NVDAObject from offset 0
DEBUGWARNING - virtualBuffers.VirtualBufferTextInfo._getNVDAObjectFromOffset (12:42:47.857) - MainThread (18632):
Couldn't get NVDAObject from offset 0
DEBUGWARNING - virtualBuffers.VirtualBufferTextInfo._getNVDAObjectFromOffset (12:42:47.857) - MainThread (18632):
Couldn't get NVDAObject from offset 0
DEBUGWARNING - browseMode.BrowseModeDocumentTreeInterceptor._set_selection (12:42:47.857) - MainThread (18632):
Invalid NVDAObjectAtStart
DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.857) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::getLineOffsets, 1068:
Offset of 0 too big for buffer, returning false

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.857) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::getLineOffsets, 1068:
Offset of 0 too big for buffer, returning false

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.867) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateTextFieldNodeAtOffset, 867:
Buffer is empty, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.867) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateControlFieldNodeAtOffset, 891:
Could not locate node at offset, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.867) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateTextFieldNodeAtOffset, 867:
Buffer is empty, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.877) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateControlFieldNodeAtOffset, 891:
Could not locate node at offset, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.881) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::findNodeByAttributes, 975:
buffer empty, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.887) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateTextFieldNodeAtOffset, 867:
Buffer is empty, returning NULL

DEBUGWARNING - RPC process 2828 (msedge.exe) (12:42:47.887) - Dummy-14 (2896):
Thread 8808, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::locateControlFieldNodeAtOffset, 891:
Could not locate node at offset, returning NULL

IO - inputCore.InputManager.executeGesture (12:42:48.264) - winInputHook (13092):
Input: kb(laptop):downArrow
IO - speech.speech.speak (12:42:48.267) - MainThread (18632):
Speaking ['Leer']
```
This is related to #12858.
